httpx/_transports/asgi.py

```diff
@@ -155,9 +155,14 @@ async def send(message: dict) -> None:
         assert status_code is not None
         assert response_headers is not None
 
-        async def response_stream() -> typing.AsyncIterator[bytes]:
-            yield b"".join(body_parts)
+        class ASGIResponseStream(AsyncByteStream):
```

Comment:
Thoughts on moving this class definition at the module level? Doesn't seem like it encloses any data from this function call (that wouldn't be ideal anyway).